### ISSUE TITLE
AB DropBox - Complete - I hope.

### ISSUE THREAD

**Issue by abojo32** [abojo32/ABDropBox#1]
@codepathreview

**Comment by abojo32** [comment]
couple of minor issues that I will have questions for during class tomorrow.

**Comment by chieger** [comment]
Nice work Anthony! 👍 

Try to challenge yourself to implement some of the optional features on the next assignment 😄 

Did you have a question about scrolling?

~ Charlie
CodePath

**Comment by abojo32** [comment]
Thanks Charlie! I will definitely try some of the optional features on the upcoming assignments. Time just got away from me this week.

I ran into an error when trying to implement scrolling on two different view controllers. When creating the outlet for each I use the same name "scrollView", using the below link as a reference. When I created the second of the view controllers and ran the simulator it would error out each time I tried to hit the second view controller I created. it was giving me "invalid redeclaration" errors. I was able to resolve by giving the outlet on the second view controller a different name.

any ideas?

thanks!

http://courses.codepath.com/courses/ios_for_designers/pages/using_uiscrollview

**Comment by chieger** [comment]
- That error typically means that you have declared 2 functions with the same name in the same class. i.e.
  ![](http://i.imgur.com/SGYaeqT.png)
- Or in this case, I would suspect that there had been 2 outlets created with the same name in the same view controller. ie.
  ![](http://i.imgur.com/Jp8ULGT.png)

If we have multiple scroll views within a view controller, we have to give them all unique names, `feedScrollView` or `menuScrollView` etc.

However, scroll views that are in different view controller classes can have the same name. In the case of the drop box mock, each view controller in the storyboard that had a scroll view would need to have it's own view controller (Cocoa touch) Swift file.

**Comment by abojo32** [comment]
In this instance, each was a separate view controller, each with their own navigation controller. I declared the UIScrollView function in each separately but gave the outlets the same exact name "scrollView", thinking it wouldn't be a problem since they were separate view controllers with their own Swift files.

i figured it wouldn't hurt to rename the outlet in the second view controller to see what happened. I named it "ScrollView" (capital S) and it worked. 

but, per your notes, it should have worked all along if the view controller and swift files were different, so I must have messed something up somewhere.

**Comment by chieger** [comment]
Yes, I believe there is something else at play here. You will often have variables, constants and functions with similar names in different view controllers. 

**BTW**, it is convention to always name **variables**(`var`), **constants** (`let`), and **functions**(func) with `lowerCamelCasing`. **Classes**, like when naming a custom view controller, will always have `UpperCamelCasing`.

**Examples:**

``` swift
 func mySweetFunction() {
}

var valueSubjectToChange = 0

let myName = "Charlie"

class MyCustomViewController: UIViewController {
}
```
